File: ai-service/app/infrastructure/mongodb/documents/business_summary_document.py
from typing import Any

from pydantic import Field

from app.domain.knowledge.entities.business_summary import BusinessSummary
from app.infrastructure.mongodb.documents.base_document import BaseMongoDocument


class BusinessSummaryDocument(BaseMongoDocument):
    """MongoDB document model representing a BusinessSummary."""

    organization_id: str = Field(default="", index=True)
    store_id: str = Field(default="", index=True)
    merchant_id: str = Field(default="")
    knowledge_version: int = Field(default=1)
    processing_status: str = Field(default="pending")
    embedding_status: str = Field(default="pending")
    summary_status: str = Field(default="pending")
    checksum: str = Field(default="")
    document_version: int = Field(default=1)
    source_type: str = Field(default="manual")
    document_id: str = Field(..., index=True)
    version_number: int = Field(default=1, ge=1)
    title: str = Field(...)
    summary: str = Field(...)
    metadata: dict[str, Any] = Field(default_factory=dict)

    def to_entity(self) -> BusinessSummary:
        return BusinessSummary(
            id=str(self.id),
            document_id=self.document_id,
            version_number=self.version_number,
            title=self.title,
            summary=self.summary,
            metadata=self.metadata,
            created_at=self.created_at,
            updated_at=self.updated_at,
        )

    @classmethod
    def from_entity(cls, entity: BusinessSummary) -> "BusinessSummaryDocument":
        return cls(
            _id=entity.id,
            document_id=entity.document_id,
            version_number=entity.version_number,
            title=entity.title,
            summary=entity.summary,
            metadata=entity.metadata,
            created_at=entity.created_at,
            updated_at=entity.updated_at,
        )

File: Back-end/AI-Sales Agent/Domain/Mongo/BusinessSummaryDocument.cs
using MongoDB.Bson.Serialization.Attributes;

namespace AI_Sales_Agent.Domain.Mongo
{
    public class BusinessSummaryDocument : MongoBaseDocument
    {
        [BsonElement("organization_id")]
        public string OrganizationId { get; set; } = string.Empty;

        [BsonElement("store_id")]
        public string StoreId { get; set; } = string.Empty;

        [BsonElement("merchant_id")]
        public string MerchantId { get; set; } = string.Empty;

        [BsonElement("knowledge_version")]
        public int KnowledgeVersion { get; set; } = 1;

        [BsonElement("processing_status")]
        public string ProcessingStatus { get; set; } = "pending";

        [BsonElement("embedding_status")]
        public string EmbeddingStatus { get; set; } = "pending";

        [BsonElement("summary_status")]
        public string SummaryStatus { get; set; } = "pending";

        [BsonElement("checksum")]
        public string Checksum { get; set; } = string.Empty;

        [BsonElement("document_version")]
        public int DocumentVersion { get; set; } = 1;

        [BsonElement("source_type")]
        public string SourceType { get; set; } = "manual";

        [BsonElement("document_id")]
        public string DocumentId { get; set; } = string.Empty;

        [BsonElement("version_number")]
        public int VersionNumber { get; set; } = 1;

        [BsonElement("title")]
        public string Title { get; set; } = string.Empty;

        [BsonElement("summary")]
        public string Summary { get; set; } = string.Empty;

        [BsonElement("metadata")]
        public Dictionary<string, object> Metadata { get; set; } = new();
    }
}
